src/renderer/services/midgard/stake.ts

```diff
@@ -29,54 +24,55 @@ const createStakeService = (
    * effects at the subscription callback as stream is completed.
    * That's why we need to create new stream via factory to have
    * a single stream per http request
-   * @example /src/renderer/views/deposit/DepositView.tsx
+   * @example /src/renderer/views/deposit/share/ShareView.tsx
    */
-  const getStakes$ = (): StakersAssetDataLD =>
+  const getStakes$ = (asset: Asset, address: Address): StakersAssetDataLD =>
     FP.pipe(
-      combineLatest([api$.pipe(map(RD.toOption)), address$, selectedPoolAsset$]),
-      map(([oApi, oAddress, oSelectedAsset]) => sequenceTOption(oApi, oAddress, oSelectedAsset)),
-      switchMap(
-        O.fold(
-          () => Rx.EMPTY,
-          ([api, address, selectedAsset]) =>
-            FP.pipe(
-              api.getMemberDetail({ address }),
-              map(({ pools }) => pools),
-              map(A.findFirst((poolDetails) => poolDetails.pool === assetToString(selectedAsset)))
-            )
-        )
-      ),
-      map(O.fold(() => RD.initial, RD.success)),
-      liveData.map((poolDetails) => ({
-        asset: poolDetails.pool,
-        assetStaked: poolDetails.assetAdded,
-        assetWithdrawn: poolDetails.assetWithdrawn,
-        dateFirstStaked: Number(poolDetails.dateFirstAdded),
-        runeStaked: poolDetails.runeAdded,
-        runeWithdrawn: poolDetails.runeWithdrawn,
-        units: poolDetails.liquidityUnits
-      })),
-      catchError((e) => {
-        /**
-         * 404 response is returned in 2 cases:
-         * 1. Pool doesn't exist at all
-         * 2. User has no any stake units for the pool
-         * In both cases return initial state as `No Data` identifier
-         */
-        if ('status' in e && e.status === 404) {
-          return Rx.of(RD.initial)
-        }
+      api$.pipe(
+        RxOp.map(RD.toOption),
+        RxOp.switchMap(
+          O.fold(
+            () => Rx.EMPTY,
+            (api) =>
+              FP.pipe(
+                api.getMemberDetail({ address }),
+                RxOp.map(({ pools }) => pools),
+                RxOp.map(A.findFirst((poolDetails) => poolDetails.pool === assetToString(asset)))
+              )
+          )
+        ),
+        RxOp.map(O.fold(() => RD.initial, RD.success)),
```

Comment:
hmmmm. i see that this is previous implementation but not sure that i fully understand what's going on😅 why so complicated ?
maybe just use `liveData` ?
```ts
 api$.pipe(
        liveData.chain((api) =>
          FP.pipe(
            api.getMemberDetail({ address }),
            RxOp.map(RD.success),
            liveData.map(({ pools }) => pools),
            liveData.chain(
              FP.flow(
                A.findFirst((poolDetails) => poolDetails.pool === assetToString(asset)),
                liveData.fromOption(() => Error('No pool'))
              )
            )
          )
        ),
        liveData.map((poolDetails) => ({...
```

Reply:
Done 2e2851c